.github/workflows/code_style_check.yml

```diff
@@ -0,0 +1,28 @@
+name: Check coding style
+
+on:
+  push:
+    paths:
+    - 'source/**'
+    - 'tests/**'
+  pull_request:
+    paths:
+    - 'source/**'
+    - 'tests/**'
+
+jobs:
+  check-style:
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    - name: set up python environment
```

Comment:
Do we need to check out more than the last commit?

Reply:
I don't know much about how checkout works. If I get the full version then the latest commit is enough; if it only gives the changes in the commit then that would be problematic. Also, I don't really know how to check out only the latest commit (or why that version doesn't do that by default).

Reply:
FYI actions/checkout checks out the last commit by default already. ~~but you could use v3 of the action (instead of v2).~~